src/engine/v3/V3Interpreter.v3

```diff
@@ -1105,6 +1113,21 @@ component V3Interpreter {
 		}
 		push(Value.V128(low, high));
 	}
+	def doExtractLane<T, S>(size: byte, view: u64 -> T, extend: T -> S, stack_push: (S) -> void) {
+		var v = Values.v_v128(pop());
+		var idx = codeptr.read1();
+		var low = v.0, high = v.1;
+		var val: S;
+		def half_lanes = (16 / size) >> 1;
+		if (idx < half_lanes) { // Extract a lane from low
+			var shift = byte.view(idx * size * 8);
```

Comment:
Here (and elsewhere), it's a tiny bit more efficient to do a `u6.view(x)` to compute a shiftor (RHS of shift), because Virgil semantics are that the shiftor is bounds-checked against the width of the type, and `u6` is always in bounds for `u64`.

Reply:
Ah that makes perfect sense since we only have 2^6 bits to shift within a `u64`. Will update accordingly.